server/backup.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package server
+
+import (
+	"bufio"
+	"context"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+func (s *Server) periodicBackup() {
+	lastWriteGen := uint64(0)
+	for {
+		gen := s.db.WriteGen()
+		if gen != lastWriteGen {
+			if err := s.doBackup(); err != nil {
+				log.Printf("Failed to take backup: %v", err)
+			} else {
+				lastWriteGen = gen
+			}
+			time.Sleep(time.Minute)
+		}
+	}
+}
+
+func (s *Server) doBackup() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	start := time.Now()
+	f, err := os.Open(s.db.Path())
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	b := bufio.NewReader(f)
```

Comment:
Even with a lot of secrets, the file will be pretty small. We probably don't need a buffer, and I'd even be tempted to `os.ReadFile` the entire thing and call `sum := md5.Sum(data)` to save a few lines. YMMV.

Reply:
Yeah, fair. Rewritten to os.ReadFile and whatnot.